RAG_supporters/dataset/domain_assessment_parser.py

```diff
@@ -0,0 +1,268 @@
+            for i, chunk in enumerate(chunks):
+                logger.info(f"Processing chunk {i+1} ({len(chunk)} rows)")
+                records.extend(self._parse_dataframe(chunk))
+        else:
+            # Read entire file
+            df = pd.read_csv(csv_path)
+            records = self._parse_dataframe(df)
+        
+        logger.info(f"Parsed {len(records)} records from {csv_path}")
+        return records
+    
+    def parse_multiple_csvs(
+        self, 
+        csv_paths: List[Union[str, Path]],
+        deduplicate: bool = True
+    ) -> List[DomainAssessmentRecord]:
+        """
+        Parse multiple CSV files and optionally deduplicate.
+        
+        Args:
+            csv_paths: List of CSV file paths
+            deduplicate: If True, removes duplicate records based on chroma IDs
+            
+        Returns:
+            Combined list of parsed records
+        """
+        all_records = []
+        
+        for csv_path in csv_paths:
+            try:
+                records = self.parse_csv(csv_path)
+                all_records.extend(records)
+            except Exception as e:
+                logger.error(f"Failed to parse {csv_path}: {e}")
+                raise
+        
+        if deduplicate:
+            all_records = self._deduplicate_records(all_records)
+        
+        logger.info(f"Total records after processing {len(csv_paths)} CSV files: {len(all_records)}")
+        return all_records
+    
+    def _parse_dataframe(self, df: pd.DataFrame) -> List[DomainAssessmentRecord]:
+        """
+        Parse DataFrame into records.
+        
+        Args:
+            df: Pandas DataFrame
+            
+        Returns:
+            List of parsed records
+        """
+        records = []
+        
+        for idx, row in df.iterrows():
+            try:
+                record = DomainAssessmentRecord.from_row(row)
+                records.append(record)
+            except Exception as e:
+                logger.warning(f"Skipping row {idx}: {e}")
+        
+        return records
+    
+    def _deduplicate_records(
+        self, 
+        records: List[DomainAssessmentRecord]
+    ) -> List[DomainAssessmentRecord]:
+        """
+        Remove duplicate records based on chroma IDs.
+        
+        Args:
+            records: List of records
+            
+        Returns:
+            Deduplicated list
+        """
+        seen_pairs = set()
+        unique_records = []
+        
+        for record in records:
+            pair_id = (record.chroma_source_id, record.chroma_question_id)
+            if pair_id not in seen_pairs:
+                seen_pairs.add(pair_id)
+                unique_records.append(record)
+        
+        removed = len(records) - len(unique_records)
+        if removed > 0:
+            logger.info(f"Removed {removed} duplicate records")
+        
+        return unique_records
+    
+    def validate_schema(self, csv_path: Union[str, Path]) -> bool:
```

Comment:
Column aliases are handled in two places:

from_row(): row.get('source_text', row.get('source', ''))
validate_schema(): column_aliases dict

Should be centralized - normalize column names once on the DataFrame.